torchdata/dataloader2/reading_service.py

```diff
@@ -188,13 +258,14 @@ class MultiProcessingReadingService(ReadingServiceInterface):
 
     def __init__(
         self,
-        num_workers: int = 0,
+        num_workers: int,
         multiprocessing_context: Optional[str] = None,
         worker_prefetch_cnt: int = 10,
         main_prefetch_cnt: int = 10,
         worker_init_fn: Optional[Callable[[DataPipe, WorkerInfo], DataPipe]] = None,
         worker_reset_fn: Optional[Callable[[DataPipe, WorkerInfo, SeedGenerator], DataPipe]] = None,
     ) -> None:
+        assert num_workers > 0, "Please use `InProcessReadingService` for num_workers=0"
```

Comment:
I am making a BC-breaking change here. `MPRS` won't accept `num_workers=0` anymore.

In previous commits, I have been trying to add a `__new__` to return `InProcessReadingService` when `num_workers=0`. However, it doesn't work well with `pickling` due to clone and mp.